subscription-fee/src/subtract_payments.rs

```diff
@@ -0,0 +1,104 @@
+use auto_farm::common::{address_to_id_mapper::AddressId, unique_payments::UniquePayments};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[must_use]
+#[derive(Debug, PartialEq, Eq, Clone, TopEncode, TopDecode, TypeAbi)]
+pub enum MyVeryOwnScResult<
+    T: NestedEncode + NestedDecode + TypeAbi,
+    E: NestedEncode + NestedDecode + TypeAbi,
+> {
+    Ok(T),
+    Err(E),
+}
+
+impl<T: NestedEncode + NestedDecode + TypeAbi, E: NestedEncode + NestedDecode + TypeAbi>
+    From<Result<T, E>> for MyVeryOwnScResult<T, E>
+{
+    fn from(value: Result<T, E>) -> Self {
+        match value {
+            Result::Ok(t) => MyVeryOwnScResult::Ok(t),
+            Result::Err(e) => MyVeryOwnScResult::Err(e),
+        }
+    }
+}
+
+impl<T: NestedEncode + NestedDecode + TypeAbi, E: NestedEncode + NestedDecode + TypeAbi>
+    MyVeryOwnScResult<T, E>
+{
+    pub fn is_err(&self) -> bool {
+        matches!(*self, MyVeryOwnScResult::Err(_))
+    }
+}
+
+#[multiversx_sc::module]
+pub trait SubtractPaymentsModule:
+    crate::fees::FeesModule
+    + crate::service::ServiceModule
+    + crate::pair_actions::PairActionsModule
+    + multiversx_sc_modules::ongoing_operation::OngoingOperationModule
+{
+    #[endpoint(subtractPayment)]
+    fn subtract_payment(
+        &self,
+        user_id: AddressId,
+        opt_specific_token: Option<EgldOrEsdtTokenIdentifier>,
```

Comment:
substract payment should receiv user_id and service_id. The subscriber should specify when registering a service the price. He can register multiple services.

So in this case on line 50 - caller as it was registered as a Service, he can have multiple services, multiple packages.

The autoClaimBoosted contract will register once as a service provider, but will register 2 services, because of the 2 prices. One user can approve multiple services from the same provider.